src/streaming/SourceBufferSink.js

```diff
@@ -79,7 +79,9 @@ function SourceBufferSink(mediaSource, mediaInfo, onAppendedCallback) {
     function reset() {
         if (buffer) {
             try {
-                mediaSource.removeSourceBuffer(buffer);
+                if (!buffer.addEmbeddedTrack) {
```

Comment:
@nicosang, what do you think if we change this condition with the following one?

`if (buffer.getClassName && buffer.getClassName() !== 'TextSourceBuffer') {`

It is two different things of doing exactly the same but I think this second approach explains itself why the if is there.

Reply:
I fully agree!